src/Uno.UI/UI/Xaml/Media/ImageBrush.wasm.cs

```diff
@@ -2,11 +2,21 @@
 using Windows.Foundation;
 using Windows.UI.Xaml.Wasm;
 using Uno.Disposables;
+using System.Threading.Tasks;
+using System.Globalization;
+using System.Collections.Generic;
+using Uno.Extensions;
+using Microsoft.Extensions.Logging;
+using Uno.Logging;
+using System.Collections.Concurrent;
 
 namespace Windows.UI.Xaml.Media
 {
 	partial class ImageBrush
 	{
+		private static readonly ConcurrentDictionary<string, Size> _naturalSizeCache = new ConcurrentDictionary<string, Size>();
```

Comment:
I didn't think I needed one either for exactly that reason but I would get errors telling me that I was trying to add an already existing key otherwise.

Reply:
The problem with the `ConcurrentDictionary` is the overhead for thread safety we don't need. If you have an existing key exception, you should find were it's coming from.